tests/core/test_snapshot.py

```diff
@@ -325,13 +325,11 @@ def test_missing_intervals_end_bounded_with_lookback(make_snapshot):
     snapshot.intervals = [(to_timestamp(start), to_timestamp(end))]
 
     execution_ts = to_timestamp("2023-01-03")
-    assert snapshot.missing_intervals(start, start, execution_time=execution_ts) == [
-        (to_timestamp(start), to_timestamp(end)),
+    assert snapshot.missing_intervals(start, start, execution_time=execution_ts) == []
```

Comment:
`start` is categorical here, which means the end is inclusive. Why wasn't `2023-01-01` returned as missing?

Reply:
because we only look at lookback if an interval is missing.